tests/datasources/test_chirps.py

```diff
@@ -0,0 +1,360 @@
+        return chirps
+
+    return _mock_chirps
+
+
+@pytest.fixture
+def mock_xr_load_dataset(mocker):
+    """Mock GeoPandas file reading function."""
+    ds = xr.Dataset()
+    return mocker.patch(
+        "aatoolbox.datasources.chirps.chirps.xr.open_mfdataset",
+        return_value=ds,
+    )
+
+
+def test_valid_arguments_class(mock_country_config):
+    """Test for resolution in initialisation class."""
+    geo_bounding_box = GeoBoundingBox(
+        lat_max=6, lat_min=3.2, lon_max=2, lon_min=-3
+    )
+    with pytest.raises(ValueError):
+        ChirpsDaily(
+            country_config=mock_country_config,
+            geo_bounding_box=geo_bounding_box,
+            resolution=0.10,
+        )
+
+
+@pytest.fixture
+def mock_download(mocker, mock_chirps):
+    """Create mock for download method."""
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.chirps.chirps._Chirps._download"
+    )
+
+    mocker.patch(
+        (
+            "aatoolbox.datasources.chirps.chirps.ChirpsMonthly"
+            "._get_last_available_date"
+        ),
+        return_value=CURRENT_DATE,
+    )
+
+    def _mock_download(
+        frequency: str,
+        start_date,
+        end_date,
+    ):
+        chirps = mock_chirps(frequency=frequency)
+        chirps.download(
+            start_date=start_date,
+            end_date=end_date,
+        )
+        args_download = download_mock.call_args_list
+        url_list = [k["url"] for (_, k) in args_download]
+        filepath_list = [k["filepath"] for (_, k) in args_download]
+        return url_list, filepath_list
+
+    return _mock_download
+
+
+def test_download_monthly(
+    mock_aa_data_dir, mock_country_config, mock_download
+):
+    """Test of call download for monthly data."""
+    url_list, filepath_list = mock_download(
+        frequency="monthly",
+        start_date=START_DATE,
+        end_date=END_DATE,
+    )
+
+    url_list_control = [
+        (
+            "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+            ".CHIRPS/.v2p0/.monthly/.global/.precipitation/"
+            f"X/%28-3.0%29%282.0%29RANGEEDGES/"
+            f"Y/%286.0%29%283.2%29RANGEEDGES/"
+            f"T/%28{MONTH_NAME}%20{START_YEAR}%29"
+            f"%28{MONTH_NAME}%20{START_YEAR}"
+            "%29RANGEEDGES/data.nc"
+        )
+        for MONTH_NAME in [
+            calendar.month_abbr[month]
+            for month in range(int(START_MONTH), int(END_MONTH) + 1)
+        ]
+    ]
+
+    filepath_list_control = [
+        mock_aa_data_dir
+        / (
+            f"public/raw/{mock_country_config.iso3}/{DATASOURCE_BASE_DIR}/"
+            f"abc_chirps_monthly_{START_YEAR}_{MONTH:02d}_"
+            "r0.05_Np6Sp3Ep2Wm3.nc"
+        )
+        for MONTH in range(int(START_MONTH), int(END_MONTH) + 1)
+    ]
+
+    assert url_list == url_list_control
+    assert filepath_list == filepath_list_control
+
+
+def test_download_daily(mock_aa_data_dir, mock_country_config, mock_download):
```

Comment:
This test needs more mocking (it still needs an internet connection), as well as `test_chirps_load`